content/articles/angular12-material-table/index.md

```diff
@@ -0,0 +1,297 @@
+    "course": "Bsc Computer Science",
+    "yearOfStudy": 4
+  }]
+```
+
+Create  `services/apiService.ts` service file and add the following:
+
+First, begin by defining the `baseURL` in the environment to make your code look organized and for reusability purposes.
+
+```typescript
+// ...
+
+export const environment = {
+  // ...
+  baseURL:'assets/',
+};
+// ...
+```
+
+Then update the service as shown below:
+
+```typescript
+// ...
+//import student intreface
+import {Student} from "../student";
+//import this to make http requests
+import {HttpClient} from "@angular/common/http";
+//we've defined our base url here in the env
+import {environment} from "../../environments/environment";
+// ....
+//class apiService
+export class ApiService {
+
+  constructor(private httpClient: HttpClient) { }
+
+  /**
+   * This method returns students details
+   */
+  getStudentsInformation(): Observable<Student[]>{
+    return this.httpClient.get<Student[]>(`${environment.baseURL}student.json`);
+  }
+}
+```
+
+Now that we've got the logic to get data from our API, let's now proceed and add this to the controller, but first, let's update our `app.module.ts` file as follows:  
+
+```typescript
+...
+
+import { HttpClientModule} from "@angular/common/http";
+
+@NgModule({
+  declarations: [
+    ...
+  ],
+  imports: [
+    ...
+    HttpClientModule, //endsure your import this module
+  ],
+  ...
+})
+...
+
+```
+
+Then update the `app.component.ts` contents as shown below:
+
+```typescript
+...
+import {MatTableDataSource} from "@angular/material/table";
+import {Student} from "./student";
+import {ApiService} from "./services/api.service";
+
+@Component({
+  ...
+})
+export class AppComponent implements OnInit{
+
+  student: Student[]=[];
+  // columns we will show on the table
+  public displayedColumns = ['firstName', 'lastName', 'studentEmail', 'yearOfStudy', 'registrationNumber', 'course'  ];
+  //the source where we will get the data
+  public dataSource = new MatTableDataSource<Student>();
+
+  //dependency injection
+  constructor(private studentApiService: ApiService) {
+  }
+
+  ngOnInit(){
+    //call this method on component load
+    this.getStudentsInformation();
+  }
+  /**
+   * This method returns students details
+   */
+  getStudentsInformation(){
+    this.studentApiService.getStudentsInformation()
+      .subscribe((res)=>{
+        console.log(res);
+      })
+  }
+}
+
+```
+
+At this point, you can now test your application by logging your API response observable output as shown below:
+
+1. Serve your application by running `ng serve --port 3000`.
+2. Go to your browser and open the new tab and enter `localhost:3000`.
+3. `Ctrl + Shift + I` to go to the logs.
+
+You will see the following output if you followed the correct steps above:  
+
+Output:
+
+[console1](/engineering-education/angular12-material-table/console.png)
+[console2](/engineering-education/angular12-material-table/exapnded-output.png)
+
```

Comment:
Please use descriptive alt text. Also, these images need a `!` at the beginning of the line to display. Do this to all your images.